python/cuml/test/utils.py

```diff
@@ -203,17 +241,53 @@ def get_models(self):
             specified custom_constructor.
         """
         classes = self._get_classes()
-        models = {name: cls for name, cls in classes
-                  if cls not in self.exclude_classes and
-                  issubclass(cls, cuml.Base)}
+        models = {
+            name: cls
+            for name,
+            cls in classes
+            if cls not in self.exclude_classes and issubclass(cls, cuml.Base)
+        }
         models.update(self.custom_constructors)
         return models
 
 
-def get_classes_from_package(package):
-    modules = [m for name, m in inspect.getmembers(package, inspect.ismodule)]
-    classes = [ClassEnumerator(module).get_models() for module in modules]
-    return {k: v for dictionary in classes for k, v in dictionary.items()}
+def get_classes_from_package(package, import_sub_packages=False):
+    """
+    Gets all modules imported in the specified package and returns a dictionary
+    of any classes that derive from `cuml.Base`
+
+    Parameters
+    ----------
+    package : python module The python module to search import_sub_packages :
+        bool, default=False When set to True, will try to import sub packages
+        by searching the directory tree for __init__.py files and importing
+        them accordingly. By default this is set to False
+
+    Returns
+    -------
+    ClassEnumerator Class enumerator for the specified package
+    """
+
+    if (import_sub_packages):
+        import os
+        import importlib
+
+        # First, find all __init__.py files in subdirectories of this package
+        root_dir = os.path.dirname(package.__file__)
```

Comment:
Why is the file-based approach necessary instead of the previous approach? Was it missing subdirs or something? Iterating over files feels a bit more error prone, so I just want to make sure I understand the reasoning

Reply:
Yes, it was missing some classes that derive from `Base` if they weren't imported by `import cuml`. For example, `cuml.experimental.decomposition.IncrementalPCA`. This approach, while not the most optimized, should at least find all of the classes.